pkg/agent/controller/traceflow/traceflow_controller.go

```diff
@@ -184,7 +196,13 @@ func (c *Controller) updateTraceflow(_, curObj interface{}) {
 func (c *Controller) deleteTraceflow(old interface{}) {
 	tf := old.(*opsv1alpha1.Traceflow)
 	klog.Infof("Processing Traceflow %s DELETE event", tf.Name)
-	c.deallocateTag(tf)
+	tfState := c.deleteTraceflowState(tf)
```

Comment:
We used to run into some race bugs, so I want not be a little careful here to change the code. Could you suggest the exact change you want - in syncTraceflow(), should I check or ignore c.traceflowLister.Get(traceflowName) returns before delete the cached state?

Reply:
In my mind it could be:
```
tf, err := c.traceflowLister.Get(traceflowName)
if err != nil {
	if apierrors.IsNotFound(err) {
		tfState := c.deleteTraceflowState(traceflowName)
		if tfState != nil {
			err := c.ofClient.UninstallTraceflowFlows(tfState.tag)
			if err != nil {
				klog.Errorf("Failed to uninstall Traceflow %s flows: %v", tf.Name, err)
			}
		}
		return nil
	}
	return err
}
```

Reply:
So, even for delete, the object is still in client cache (can be got from traceflowLister)?

Changed. Check.

Reply:
No, after it's deleted, Lister doesn't have it, that's why we should check if the error is "NotFound". I don't see we need the Traceflow object to remove the traceflow state cache. I think it only needs the name of the traceflow?

Reply:
Ok. Updated.